nomad/client_rpc.go

```diff
@@ -0,0 +1,123 @@
+package nomad
+
+import (
+	"fmt"
+	"time"
+
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/hashicorp/yamux"
+)
+
+// nodeConnState is used to track connection information about a Nomad Client.
+type nodeConnState struct {
+	// Session holds the multiplexed yamux Session for dialing back.
+	Session *yamux.Session
+
+	// Established is when the connection was established.
+	Established time.Time
```

Comment:
Might be better to turn this into UTC unixnanos at creation time to avoid subtle bugs

Reply:
The time object should carry this information over an RPC as it implements MarshalBinary.